### ISSUE TITLE
Generally allow initialize=true for all coupling schemes and all participants

### ISSUE THREAD

**Issue by BenjaminRodenberg** [precice/precice#1367]
**Please describe the problem you are trying to solve.**

Currently `initialize=true` is forbidden for data that is sent from first to second in a serial coupling scheme (because this data would be ignored). From [our documentation](https://precice.org/couple-your-code-initializing-coupling-data.html):

> In a serial coupling, only the second participant can send data inside initializeData(). In parallel coupling, both participants can initialize data.

I think that this restriction is basically motivated from the technical point of view, but recent changes that I implemented to develop waveform iteration remove the technical reasons (waveform iteration needs to allow initialization in all directions for serial coupling schemes. See #1352).

From the usability perspective I think this restriction makes things more complicated. It makes it more explicit that initial data sent from the first participant to the second is ignored, but this will become less relevant with waveform iteration, anyway.

**Describe the solution you propose.**

Remove the error that is raised during configuration time:

https://github.com/precice/precice/blob/67b0e6d9c9629b69d923cc540056bfd63a258a56/src/cplscheme/config/CouplingSchemeConfiguration.cpp#L987-L990

The remaining necessary technical changes should already be included in #1352.

**Describe alternatives you've considered**

Keep it as it is, but I think this makes the implementation and documentation unnecessarily complicated.

**Comment by uekerman** [comment]
Makes sense. If not too complicated maybe we could raise a warning if a user initializes data, which is never used.
(Serial cpl, first participant sends, no waveform relaxation)

**Comment by BenjaminRodenberg** [comment]
> Makes sense. If not too complicated maybe we could raise a warning if a user initializes data, which is never used. (Serial cpl, first participant sends, no waveform relaxation)

Raising a warning should not be a problem from the technical side (basically we currently raise an error in this situation. Replacind the error with a warning is possible.

But: Is this situation really a problem? I think it makes the config more flexible w.r.t the choice of the coupling scheme, if one can just always keep initialize=true everywhere (if initial data is available, why not just give it always to preCICE and let preCICE decide what to do with it?). The only reason for raising a warning is to avoid that a user believes the data is used even though it is not. But this is also a information a user can find in the documentation, if (s)he is wondering why the data does not look as expected. To me this feels like a bit too much of caution.

Additionally: Raising the warning is technically easy, but still complicates the code with some `if`s.

**Comment by BenjaminRodenberg** [comment]
#1368 now throws a warning, if a serial coupling with waveform order zero is used and `initialize="true"` for initial data sent by the first participant.
